deploy.yml

```diff
@@ -16,10 +16,34 @@
   remote_user: "{{ web_user }}"
 
   vars:
-    deploy_build_before: "{{ playbook_dir }}/deploy-hooks/build-before.yml"
-    deploy_build_after: "{{ playbook_dir }}/roles/deploy/hooks/build-after.yml"
-    deploy_finalize_before: "{{ playbook_dir }}/roles/deploy/hooks/finalize-before.yml"
-    deploy_finalize_after: "{{ playbook_dir }}/roles/deploy/hooks/finalize-after.yml"
+
+    #deploy_before:
+    #deploy_initialize_before:
+    #deploy_initialize_after:
+    #deploy_update_before:
+    #deploy_update_after:
+    #deploy_prepare_before:
+    #deploy_prepare_after:
+
+    deploy_build_before:
+      - "{{ playbook_dir }}/deploy-hooks/build-before.yml"
+
+    deploy_build_after:
+      #- "{{ playbook_dir }}/deploy-hooks/sites/{{ site }}-build-after.yml"
+      - "{{ playbook_dir }}/roles/deploy/hooks/build-after.yml"
+
+    #deploy_share_before:
+    #deploy_share_after:
+
+    deploy_finalize_before:
+      - "{{ playbook_dir }}/roles/deploy/hooks/finalize-before.yml"
+
+    deploy_finalize_after:
+      - "{{ playbook_dir }}/roles/deploy/hooks/finalize-after.yml"
+
+    #deploy_after:
+
+
```

Comment:
@strarsis Your adjustments to these `vars` in `deploy.yml` were helpful in initiating internal discussion and the decision to move the deploy role's vars into `roles/deploy/defaults/main.yml`. This will help `deploy.yml` be more concise. It will be more consistent with how other roles have their vars in their respective `defaults/main.yml` files.

Could you help us take the first step toward this goal in this PR by moving just the deploy hook vars out of `deploy.yml` to a new section at the end of `roles/deploy/defaults/main.yml`? I think you could start the section with a link to the deploy hooks docs, where all hooks are listed in sequence. Then I think it would be preferable to omit the unused/commented hook variables here in `roles/deploy/defaults/main.yml`, unless there's a strong argument for duplication.

Example
```yml
# roles/deploy/defaults/main.yml
...

# Deploy hooks
# For list of hooks and explanation, see https://roots.io/trellis/docs/deploys/#hooks
deploy_build_before:
  - "{{ playbook_dir }}/deploy-hooks/build-before.yml"

deploy_build_after:
  - "{{ playbook_dir }}/roles/deploy/hooks/build-after.yml"
  # - "{{ playbook_dir }}/deploy-hooks/sites/{{ site }}-build-after.yml"

deploy_finalize_before:
  - "{{ playbook_dir }}/roles/deploy/hooks/finalize-before.yml"

deploy_finalize_after:
  - "{{ playbook_dir }}/roles/deploy/hooks/finalize-after.yml"
```